agent/consul/state/config_entry.go

```diff
@@ -555,6 +651,44 @@ func testCompileDiscoveryChain(
 	return chain.Protocol, chain.Nodes[chain.StartNode], nil
 }
 
+func (s *Store) ServiceDiscoveryChain(
+	ws memdb.WatchSet,
+	serviceName string,
+	entMeta *structs.EnterpriseMeta,
+	req discoverychain.CompileRequest,
+) (uint64, *structs.CompiledDiscoveryChain, error) {
+
```

Comment:
Did you want all of these to use the same memdb txn?

Reply:
Yeah now there's a single transaction for the call to ServiceTopology.